StoryMap/StoryMap/Screens/Map/Coordinator/MapCoordinator.swift

```diff
@@ -7,62 +7,93 @@
+	private var subscribers = Set<AnyCancellable>()
+	
+	private var photoManagerSubscriber: AnyCancellable?
+	
+	init(presenter: UINavigationController) {
+		self.presenter = presenter
+	}
     
-    func start(_ presentFrom: UIViewController?) {
+    func start() {
         let viewModel = MapViewModel()
         let viewController = MapViewController(
             viewModel: viewModel,
             locationManager: LocationManager()
         )
         viewController.modalPresentationStyle = .fullScreen
-        viewModel.onAddStory = { [weak self] location in
-            self?.showAddStory(with: location)
-        }
-        viewModel.onOpenStory = { [weak self] story in
-            self?.showStoryDetail(with: story)
-        }
-        viewModel.onOpenStoryList = { [weak self] in
-            self?.showStoryList()
-        }
-		onDeleteStory = viewModel.storyDeleted
+        viewModel.addStorySubject
+			.sink { [weak self] location in
+				self?.showAddStory(with: location)
+			}
+			.store(in: &subscribers)
+        viewModel.openStorySubject
+			.sink { [weak self] story in
+				self?.showStoryDetail(with: story)
+			}
+			.store(in: &subscribers)
+        viewModel.openStoryListSubject
+			.sink { [weak self] in
+				self?.showStoryList()
+			}
+			.store(in: &subscribers)
+		
+		deleteStorySubject = viewModel.storyDeletedSubject
         presenter.pushViewController(viewController, animated: true)
     }
     
-    func stop() {
-        presenter.popViewController(animated: true)
-    }
-    
     private func showAddStory(with location: Location) {
-        addStoryCoordinator = AddStoryCoordinator(location: location, simple: true)
-        addStoryCoordinator?.onShowStory = { [weak self] story in
-            self?.showStoryDetail(with: story)
-        }
-        addStoryCoordinator?.start(presenter.topViewController)
+		if photoManagerSubscriber != nil {
+			photoManagerSubscriber?.cancel()
+			photoManagerSubscriber = nil
+		}
+		
+		photoManagerSubscriber = photoManager.imageSubject
+			.receive(on: DispatchQueue.main)
+			.sink { [weak self] image in
+				StoryDataProvider.shared.createStory(
+					from: image,
+					and: location
+				)
+				self?.presenter.dismiss(animated: true)
+			}
+		
+		let viewController = photoManager.makeViewController(with: .camera)
+		presenter.present(viewController, animated: true)
     }
     
     private func showStoryDetail(with story: Story) {
-        storyDetailCoordinator = StoryDetailCoordinator(story: story)
-		storyDetailCoordinator?.onDeleteStory = { [weak self] story in
-			self?.onDeleteStory?(story)
+		if storyDetailCoordinator != nil {
+			storyDetailCoordinator = nil
 		}
-        storyDetailCoordinator?.presenter = presenter
-        storyDetailCoordinator?.start(nil)
+		
+		storyDetailCoordinator = StoryDetailCoordinator(presenter: presenter, story: story)
+		storyDetailCoordinator?.deleteStorySubject
+			.sink { [weak self] story in
+				self?.deleteStorySubject?.send(story)
```

Comment:
DeleteStory is passed from DetailCoordinator -> MapCoordinator - > MapViewModel
Correct? Hard to follow...